internal/extproc/watcher.go

```diff
@@ -66,9 +68,54 @@ func (cw *configWatcher) loadConfig() error {
 	}
 	cw.lastMod = stat.ModTime()
 	cw.l.Info("loading a new config", slog.String("path", cw.path))
+
+	// Re-hydrate the current config file for later diffing.
+	previous := cw.current
+	current, err := cw.getConfig()
+	if err != nil {
+		return fmt.Errorf("failed to read the config file: %w", err)
+	}
+
+	// Print the diff between the old and new config.
+	cw.diff(previous, current)
+
 	cfg, err := filterconfig.UnmarshalConfigYaml(cw.path)
 	if err != nil {
 		return err
 	}
 	return cw.rcv.LoadConfig(cfg)
 }
+
+func (cw *configWatcher) getConfig() (string, error) {
+	currentByte, err := os.ReadFile(cw.path)
+	if err != nil {
+		return "", err
+	}
+	current := string(currentByte)
+	cw.current = current
+
+	return current, nil
+}
+
+func (cw *configWatcher) diff(oldConfig, newConfig string) {
```

Comment:
could you write a very simple unit test for this function so that we can what the expected output looks like?

Reply:
I’m happy to write a new unit test, but I’m unsure if it’s necessary since TestStartConfigWatcher already has a section that updates the config, triggering the diff. Additionally, TestStartConfigWatcher has been updated to reflect this behavior. My concern is that adding another test might result in an overlap with what TestStartConfigWatcher already covers. However, if you think it’s worth it, I’m open to writing the test. Please let me know your thoughts. Thanks!

Reply:
yeah please go ahead and write a test

Reply:
especially to ensure the logic in `log.Int("line", i+1), slog.String("path", cw.path), slog.String("old", oldLine), slog.String("new", newLine)` which i believe is not checked in the current test.